T/Torch/build_tarballs.jl

```diff
@@ -3,60 +3,263 @@
+filter!(!Sys.isfreebsd, platforms) # Build fails: Clang v12 crashes compiling aten/src/ATen/native/cpu/MaxUnpoolKernel.cpp.
+
+mkl_platforms = [
+    Platform("x86_64", "Linux"),
+    Platform("i686", "Linux"),
+    Platform("x86_64", "MacOS"),
+    Platform("x86_64", "Windows"),
 ]
 
+blis_platforms = filter(p -> p ∉ mkl_platforms, [
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("x86_64", "freebsd"),
+    Platform("aarch64", "macos"),
+    Platform("x86_64", "macos"),
+    Platform("x86_64", "windows"),
+])
+
+openblas_platforms = filter(p -> p ∉ union(mkl_platforms, blis_platforms), platforms)
+
+cuda_platforms = [
+    Platform("x86_64", "Linux"; cuda = "10.2"),
+    Platform("x86_64", "Linux"; cuda = "11.3"),
+]
+for p in cuda_platforms
+    push!(platforms, p)
+end
+
+platforms = expand_cxxstring_abis(platforms)
+mkl_platforms = expand_cxxstring_abis(mkl_platforms)
+blis_platforms = expand_cxxstring_abis(blis_platforms)
+openblas_platforms = expand_cxxstring_abis(openblas_platforms)
+cuda_platforms = expand_cxxstring_abis(cuda_platforms)
+
 # The products that we will ensure are always built
 products = [
-    LibraryProduct("libdoeye_caml", :libdoeye_caml, dont_dlopen = true),
-    LibraryProduct("libtorch", :libtorch, dont_dlopen = true),
+    LibraryProduct(["libtorch", "torch"], :libtorch),
+    LibraryProduct(["libtorch_cpu", "torch_cpu"], :libtorch_cpu),
 ]
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    BuildDependency(PackageSpec(name="CUDA_full_jll", version=v"10.1.243")),
-    Dependency(PackageSpec(name="CompilerSupportLibraries_jll")),
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
+    Dependency("blis_jll"; platforms = blis_platforms),
+    Dependency("CPUInfo_jll", v"0.0.20201217"),
+    Dependency("CUDNN_jll", v"8.2.4"; compat = "8", platforms = cuda_platforms),
+    Dependency("Gloo_jll", v"0.0.20210521"; platforms = filter(p -> nbits(p) == 64, platforms)),
+    Dependency("LAPACK_jll"; platforms = openblas_platforms),
+    Dependency("MKL_jll"; platforms = mkl_platforms),
+    BuildDependency("MKL_Headers_jll"; platforms = mkl_platforms),
+    Dependency("OpenBLAS_jll"; platforms = openblas_platforms),
+    Dependency("PThreadPool_jll", v"0.0.20210414"),
+    Dependency("SLEEF_jll", v"3.5.2"),
+#    Dependency("TensorRT_jll"; platforms = cuda_platforms), # Building with TensorRT is not supported: https://github.com/pytorch/pytorch/issues/60228
+    Dependency("XNNPACK_jll", v"0.0.20210622"),
+    BuildDependency(PackageSpec("protoc_jll", Base.UUID("c7845625-083e-5bbe-8504-b32d602b7110"), v"3.13.0")),
+    HostBuildDependency(PackageSpec("protoc_jll", Base.UUID("c7845625-083e-5bbe-8504-b32d602b7110"), v"3.13.0")),
```

Comment:
When you specify the version of a runtime dependency you typically also want to specify the compat (or actually only that one since the build version is automatically inferred from the lowest compatible version)